src/components/ProductList/ProductList.js

```diff
@@ -1,41 +1,87 @@
 import React, {useState, useEffect} from 'react';
 import axios from 'axios';
+import ReactPaginate from 'react-paginate';
 import './ProductList.scss';
 import OfferBox from 'components/OfferBox/OfferBox';
+import Sorting from 'components/Sorting/Sorting';
 import {baseURL} from 'helpers/baseURL.js'
 
 const ProductList = () => {
-
-  const [products, setProducts] = useState([])
+  const [products, setProducts] = useState([]);
+  const [loading, setLoading] = useState(false);
+  const [error, setError] = useState(null);
 
   useEffect(() => {
-      axios.get(baseURL)
-        .then(res => {setProducts(res.data)})
-        .catch(err => console.log(err));
+    setLoading(true);
+
+    axios.get(baseURL)
+      .then(res => {
+        setProducts(res.data);
+        setLoading(false);
+        setError(null);
```

Comment:
nie musisz tutaj nulla ustawiać, jak się nic nie zmieni to null zostaje jako default :)